### DISCUSSION TITLE
run processes one after the other

### DISCUSSION THREAD

**Discussion by hmassalha** [nextflow-io/nextflow#2807] [Q&A] [answered]
hi, 

I have three processes (S1, S2, S3) and I want them to run one after the other (S1 > S2 > S3). Each process uses a custom `jupyter-nbconvert`  code to make some analysis and generate a file. I looked into the tutorial and I didn't see a direct implementation of what I am looking for. I know it is me :) 

After several trials I set the output argument of the first process to `file 'output_s1' into result_s1` but I got an error says `Missing output file(s) "stdout_s1" expected by process "S1 (1)"`. In the tutorial, the outputed file declared in the process, but in my case I have .ipynb file and there is no direct declaration of the output file. 

I made sure my .ipynb files are working with no bugs, so I need a flag to tell a process when to start. Any suggestions ? 

Thanks, HM 

EDIT:
I looked into "[Mock dependency](https://nextflow-io.github.io/patterns/index.html#_mock_dependency)".. it seems exactly what I am looking for, I test it, it seems the output of the first process is triggered before it finished, as the process is executed via job and the nextflow run command via job too.

**Comment by bentsherman** [comment, marked as answer]
Hi @hmassalha , I'm marking your discussion as answered. Solution was to define mock dependencies between each process so that they are triggered sequentially. Additionally, make sure that your output declaration matches the name of the actual output file in your process.

**Comment by Kibubu** [comment]
Mock dependency is now called [State dependency](https://nextflow-io.github.io/patterns/index.html#_state_dependency)
